File: library/modules/tools/qtapplication/source/qtapplication.cpp

#include "tiki/qtapplication/qtapplication.hpp"

namespace tiki
{
	QtApplication::QtApplication( int argc, char** argv )
		: QApplication( argc, argv )
	{

	}

	QtApplication::~QtApplication()
	{

	}

	int QtApplication::runInternal()
	{
		return 0;
	}
}

File: engine/modules/tool/qtapplication/qtapplication.lua
-- library/modules/runtime/qtapplication

local module = Module:new( "qtapplication" );

module:add_files( "include/**/*.*" );
module:add_include_dir( "include" );
module:add_files( "*.lua" );
module:add_files( "source/*.*" );

module:add_dependency( "core/base" );
module:add_dependency( "thirdparty/qt" );

module.import_func = function(project)
	module:add_moc_file( "include/tiki/qtapplication/qtmainwindow.hpp" );
	module:add_moc_file( "include/tiki/qtapplication/qtribbontab.hpp" );
	module:add_moc_file( "include/tiki/qtapplication/qtribbonbutton.hpp" );
	module:add_moc_file( "include/tiki/qtapplication/qtribbongroup.hpp" );
end